papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/service/SprintsRestService.java

```diff
@@ -86,57 +87,60 @@
-        }
-
-        Sprint changedSprint = sprintService.closeSprint(sprint, timeBurned, timePlanned, timeNow);
-        return ResponseEntity.ok(sprintMapper.mapToApi(changedSprint));
+    private boolean isInOverlapWithLastSprint(Project project, LocalDateTimePeriod durationPeriod) {
+        Optional<Sprint> lastSprintOpt = sprintService.getLastInProject(project.getId());
+        return lastSprintOpt.isPresent() && durationPeriod.isBefore(lastSprintOpt.get().getDurationPeriod());
     }
 
+    @SuppressWarnings({"MethodLength"})
     public ResponseEntity<SprintSummaryDto> getSprintSummary(Long projectId, Long sprintId) {
-        Sprint sprint = getValidSprint(sprintId);
         Project project = projectsRestService.getValidProject(projectId);
 
         if (!project.isAdmin(userContext.getUser())) {
             throw new ForbiddenAccessException();
         }
 
-        List<Availability> availabilities = sprint.getAvailabilities();
+        Sprint currentSprint = getValidSprint(sprintId);
+        List<Availability> availabilities = currentSprint.getAvailabilities();
         List<UserAvailabilityDto> userAvailabilityDtos = getUserAvailabilityDtos(project, availabilities);
 
-        Duration totalAvailableTime = availabilities.stream()
+        Duration totalDeclaredTime = availabilities.stream()
                 .map(Availability::getTimeAvailable)
                 .reduce(Duration.ZERO, Duration::plus);
 
-        // to be expanded in issue IO2019TEAM-100
-        double coefficient = Optional.ofNullable(sprint.getTimeBurned())
-                .filter(timeBurned -> !Duration.ZERO.equals(timeBurned))
-                .map(timeBurned -> (double) sprint.getTimePlanned().toMinutes() / timeBurned.toMinutes())
-                .orElse(0.d);
-
-        return ResponseEntity.ok(createSprintSummaryDto(userAvailabilityDtos, totalAvailableTime, coefficient));
+        return ResponseEntity.ok(createSprintSummaryDto(
+                currentSprint,
+                userAvailabilityDtos,
+                totalDeclaredTime));
     }
 
-    private SprintSummaryDto createSprintSummaryDto(List<UserAvailabilityDto> userAvailabilityDtos,
-            Duration totalAvailableTime, double coefficient) {
-        return new SprintSummaryDto().membersAvailability(userAvailabilityDtos)
-                .totalAvailableTime(totalAvailableTime.toMinutes())
-                .sprintCoefficient(coefficient);
+    private SprintSummaryDto createSprintSummaryDto(Sprint currentSprint,
+            List<UserAvailabilityDto> userAvailabilityDtos, Duration totalDeclaredTime) {
+
+        double prevAverageSprintCoefficient = sprintService.getPrevSprintAverageCoefficient(currentSprint);
+        double currentAverageSprintCoefficient = currentSprint.getStats().getAverageCoefficientCache();
+
+        Long timeToAssign = (long) (totalDeclaredTime.toMinutes() / prevAverageSprintCoefficient);
```

Comment:
`(currentAverageSprintCoefficient != null && currentAverageSprintCoefficient != 0d)`?

Reply:
`Long` -> `long`?

Reply:
te Long -> long dokładniej zaraz ogarne i pozmieniam

co do dzielenia przez 0, to nie przejmowałbym się tym jeśli mamy liczby zmiennoprzecinkowe, ta operacja jest dobrze zdefiniowana a i tak ciężko powiedzieć co miałoby się wtedy dziać, bo zero tutaj oznaczałoby już problemy przy wyliczaniu współczynnika.
Jeśli bym coś zmieniał to ustalił minimalną i maksymalną wartość współczynnika przy jego wyliczaniu, jeśli wyjdzie NaN (bo chyba możemy teraz mieć 0/0), albo +-inf albo poza przedziałem, to przytne (przy czym byłby to wartości rzędu 10e-6, 10e6

Reply:
a tak bardzo chciałem kiedyś tylko na dodatnie pozwalać żeby zer nie było xd

Reply:
> a tak bardzo chciałem kiedyś tylko na dodatnie pozwalać żeby zer nie było xd

I to spoko jest docelowo, tylko potem trzeba sklejać front i backend taśmą, bo nie idzie nam idealnie robienie tego równo